packages/x-date-pickers/src/TimeField/tests/editing.TimeField.test.tsx

```diff
@@ -385,6 +389,7 @@ describe('<TimeField /> - Editing', () => {
 
         userEvent.keyPress(input, { key: 'a', ctrlKey: true });
         userEvent.keyPress(input, { key: 'Backspace' });
+        userEvent.keyPress(input, { key: 'ArrowLeft' });
```

Comment:
You are able to write immediately since when all sections are selected the editing is applied on the 1st one.

I'm putting it here because in the test I prefer to avoid using the editing when everything is selected.
The reason is that the string passed to the onChange is less readable.
But I could remove this line from the test and fire 


```tsx
fireEvent.change(input, { target: { value: '3' } }); // Press "3"
```

That would work just fine.

____

About the idea of selecting only the 1st section after cleaning, I saw your message in the issue and I think that's worth exploring. It's not related to this PR which only fix one specific bug :+1: